File: Java/api/org/rr0/im/business/event/circumstance/WeatherCondition.java
package org.rr0.im.business.event.circumstance;

/**
 * 
 *
 * @author Jérôme Beau
 * @version 10 mai 2003 16:59:55
 */
public interface WeatherCondition {
    /**
     * @associates <{org.rr0.im.business.event.circumstance.Clouds}>
     */
    Clouds getClouds();

    /**
     * @associates <{org.rr0.im.business.event.circumstance.Temperature}>
     */
    Temperature getTemperature();

    /**
     * @associates <{org.rr0.im.business.event.circumstance.Wind}>
     */
    Wind getWind();

    /**
     * @associates <{org.rr0.im.business.event.circumstance.Precipitations}>
     */
    Precipitations getPrecipitations();
}

File: api/src/org/rr0/im/business/event/circumstance/WeatherCondition.ts
import {Clouds} from "./Clouds";
import {Temperature} from "./Temperature";
import {Wind} from "./Wind";
import {Precipitations} from "./Precipitations";

export interface WeatherCondition {
  getClouds(): Clouds;

  getTemperature(): Temperature;

  getWind(): Wind;

  getPrecipitations(): Precipitations;
}
